### ISSUE TITLE
How to test model on a single image

### ISSUE THREAD

**Issue by smehida** [fizyr/keras-retinanet#220]
Hi,
How can I test model on new image? what kind of preprocessing I need to pass the image to the network?
Is this correct?
```
image = cv2.imread('test/1.jpg', 1)

draw = image.copy()
draw = cv2.cvtColor(draw, cv2.COLOR_BGR2RGB)
image, scale = resize_image(image)
image = image.astype('float32')
image[:, :, 0] -= 103.939
image[:, :, 1] -= 116.779
image[:, :, 2] -= 123.68
image = image/np.max(image)
start = time.time()
_, _, detections = model.predict_on_batch(np.expand_dims(image, axis=0))
print detections
print("processing time: ", time.time() - start)
predicted_labels = np.argmax(detections[0, :, 4:], axis=1)
scores = detections[0, np.arange(detections.shape[1]), 4 + predicted_labels]
detections[0, :, :4] /= scale
```

in This case all of my predicted labels are zero. I don't know it is for bad training or wrong test code.

**Comment by hgaiser** [comment]
The only problem is that you don't need to rescale the values to [0, 1] (in other words, no need for `image = image/np.max(image)`). Other than that it looks fine. I'll see if I can add an example like this to the jupyter notebook, since it is a more frequent question.

Can you let me know if that solves your issue ?

**Comment by smehida** [comment]
yeah Thanks my problem is solved. I found out my model is not trained very well. How many epochs roughly needed to train this model from scratch? (I thought ResNet parameters are not trainable but they are, so the number of trainable parameters are 35M which is very large for my 20000-image dataset, should I make the trainable flag of that model false?)

**Comment by hgaiser** [comment]
20.000 images sounds very reasonable, depending on your number of classes. VOC is only 12k if I remember correctly. I've also trained on much smaller datasets (< 1000 images, 2 classes). How many iterations your network needs is hard to guess, it depends on many things. My advice is to keep an eye on the loss and see if it stops decreasing.

**Comment by smehida** [comment]
Thanks for your answer. I also have 2 classes. What's the reasonable amount of the overall loss? 
currently my losses after 59 epochs (each epoch consists of 20000 steps) are like below:
 loss: 0.13 , regression loss: 0.09, classification loss: 0.04

**Comment by hgaiser** [comment]
That's hard to say, it depends on how difficult your dataset is really. But those loss values seem very reasonable to me.

I'll close this since the original issue was resolved. For further discussions, please join the Slack channel.
